src/write/mod.rs

```diff
@@ -184,16 +185,40 @@ fn yaw_field<W: Write>(movement: &Option<AutoMovement>) -> impl SerializeFn<W> +
     move |out: WriteContext<W>| match movement {
         None => string("-")(out),
         Some(AutoMovement::SetYaw(yaw)) => display(yaw)(out),
-        Some(AutoMovement::Strafe(StrafeSettings { type_, dir })) => match type_ {
-            StrafeType::ConstYawspeed(yawspeed) => display(yawspeed)(out),
-            _ => match dir {
-                StrafeDir::Yaw(yaw) => display(yaw)(out),
-                StrafeDir::Point { x, y } => tuple((display(x), string(" "), display(y)))(out),
-                StrafeDir::Line { yaw } => display(yaw)(out),
-                StrafeDir::LeftRight(count) | StrafeDir::RightLeft(count) => display(count)(out),
-                _ => string("-")(out),
-            },
-        },
+        Some(AutoMovement::Strafe(StrafeSettings { type_, dir })) => {
+            let is_constant_yawspeed = matches!(type_, StrafeType::ConstYawspeed(_));
+
+            let out = match dir {
+                StrafeDir::Yaw(yaw) => display(yaw)(out)?,
+                StrafeDir::Point { x, y } => tuple((display(x), string(" "), display(y)))(out)?,
+                StrafeDir::Line { yaw } => display(yaw)(out)?,
+                StrafeDir::LeftRight(count) | StrafeDir::RightLeft(count) => display(count)(out)?,
+                // Left or Right
+                _ => {
+                    // Write nothing for Left or Right because we will write the yawspeed in the next step.t
+                    if is_constant_yawspeed {
+                        string("")(out)?
+                    } else {
+                        string("-")(out)?
+                    }
+                }
+            };
+
+            let out = match type_ {
+                StrafeType::ConstYawspeed(yawspeed) => display(yawspeed)(out)?,
+                StrafeType::MaxAccelYawOffset(start, target, accel) => tuple((
+                    string(" "),
+                    display(start),
+                    string(" "),
+                    display(target),
+                    string(" "),
+                    display(accel),
+                ))(out)?,
+                _ => string("")(out)?,
```

Comment:
Same, does this work?
```suggestion
                _ => out,
```